components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/handlers/CibaResponseTypeHandler.java

```diff
@@ -0,0 +1,113 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.ciba.handlers;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.oauth.ciba.common.AuthReqStatus;
+import org.wso2.carbon.identity.oauth.ciba.dao.CibaDAOFactory;
+import org.wso2.carbon.identity.oauth.ciba.exceptions.CibaCoreException;
+import org.wso2.carbon.identity.oauth.dto.OAuthErrorDTO;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.authz.OAuthAuthzReqMessageContext;
+import org.wso2.carbon.identity.oauth2.authz.handlers.AbstractResponseTypeHandler;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AuthorizeReqDTO;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AuthorizeRespDTO;
+import org.wso2.carbon.identity.oauth2.model.OAuth2Parameters;
+
+/**
+ * Handles authorize requests with CibaAuthCode as response type.
+ */
+public class CibaResponseTypeHandler extends AbstractResponseTypeHandler {
+
+    private static Log log = LogFactory.getLog(CibaResponseTypeHandler.class);
+
+    @Override
+    public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx) throws IdentityOAuth2Exception {
+
+        OAuth2AuthorizeRespDTO respDTO = new OAuth2AuthorizeRespDTO();
+        OAuth2AuthorizeReqDTO authorizationReqDTO = oauthAuthzMsgCtx.getAuthorizationReqDTO();
+        try {
+            // Assigning authenticated user for the request that to be persisted.
+            AuthenticatedUser cibaAuthenticatedUser = authorizationReqDTO.getUser();
+
+            // Assigning the authentication status that to be persisted.
+            Enum authenticationStatus = AuthReqStatus.AUTHENTICATED;
+
+            String authCodeKey =
+                    CibaDAOFactory.getInstance().getCibaAuthMgtDAO().getCibaAuthCodeKey(authorizationReqDTO.getNonce());
+
+            // Update successful authentication.
+            CibaDAOFactory.getInstance().getCibaAuthMgtDAO()
+                    .persistAuthenticationSuccess(authCodeKey, cibaAuthenticatedUser);
+
+            // Building custom CallBack URL.
+            String callbackURL = authorizationReqDTO.getCallbackUrl() + "?authenticationStatus=" + authenticationStatus;
+            respDTO.setCallbackURI(callbackURL);
+            return respDTO;
+        } catch (CibaCoreException e) {
+            throw new IdentityOAuth2Exception("Error occurred in persisting authenticated user and authentication " +
+                    "status for the request made by client: " + authorizationReqDTO.getConsumerKey(), e);
+        }
+    }
+
+    @Override
+    public OAuthErrorDTO handleUserConsentDenial(OAuth2Parameters oAuth2Parameters) {
+
+        OAuthErrorDTO oAuthErrorDTO = new OAuthErrorDTO();
+        String authReqID = oAuth2Parameters.getNonce();
+        String authCodeKey = null;
```

Comment:
Is this null assignment necessary?

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1701